### PR TITLE
Login and Signup Authentication

### PR DESCRIPTION
**Pull request by AbhishekkGautam**
### Login page 
- login with `email` & `password`.
- save `token` in localStorage for persistence auth.
- `error` message if email & password do not match.

### Signup Page
- create a new account with `name`, `email` & `password`.
- confirm password input for re-checking the password.
- after signing up, the user will be redirected to the login page.

### Logout button
- on clicking the logout button, the token will be deleted from localStorage & the user will be redirected to the home page.
- and the logout button will convert into the login button. 

###  Protected Routes. 
- user cannot visit `cart` & `wishlist` pages without logging in. 
- once the user is logged in, he/she can't visit the `login` & `signup` pages. Users will need to logout to access the login page.

### CODE REVIEW

File: src/pages/Signup/Signup.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  return (
+    <main class="main-wrapper">
+      <section class="signup-section signup-container">
+        <div class="card signup-card-container">
+          <div class="card-title">Signup</div>
+          {errorMsg !== "" ? (
+            <div className="signup-error-message">{errorMsg}</div>
+          ) : null}
+          <div class="card-body">
+            <form onSubmit={e => e.preventDefault()}>
```
Comment (Sen-442b) [inline src/pages/Signup/Signup.js:50]: Any specific reason to not  use the `onSubmit` event for handling form submit?

File: src/pages/Signup/Signup.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                    setIsPasswordMatched(password === e.target.value);
+                  }}
+                />
+                {!isPasswordMatched && (
+                  <div class="error-message">Password doesn't match.</div>
+                )}
+              </div>
+              <div class="card-extra-content">
+                <div class="checkbox-group">
+                  <label class="checkbox-label" for="checkbox1">
```
Comment (Sen-442b) [inline src/pages/Signup/Signup.js:121]: You can replace it with `htmlFor`

**Review by Sen-442b — APPROVED** [review]
- The file structure is organized 
- Code looks clean to me 
-Consistent naming convention followed

File: src/context/AuthContext.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      const { status } = await signupService(
+        firstName,
+        lastName,
+        email,
+        password
+      );
+      if (status === 200 || status === 201) {
+        navigate("/login");
+      }
+    } catch (error) {
```
Comment (poornima-alugubelly) [inline src/context/AuthContext.js:46]: Nice error Handling

**Review by poornima-alugubelly — APPROVED** [review]
As mentioned , good file structure and naming 👍

File: src/pages/Signup/Signup.js (hunk 1 above)
Comment (AbhishekkGautam) [reply src/pages/Signup/Signup.js:50]: No, i just want to put login handler in onClick of button.

**Pull request merged**